prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java

```diff
@@ -67,24 +66,37 @@ public AbstractCommand(CliConsole console, ActionFactory actionFactory) {
     }
 
     protected static Path determineInstallationDirectory(Optional<Path> directoryOption) throws ArgumentParsingException {
-        Path installationDirectory = directoryOption.orElse(currentDir()).toAbsolutePath();
-        verifyDirectoryContainsInstallation(installationDirectory);
-        return installationDirectory;
+        return determineInstallationDirectory(directoryOption, currentDir());
     }
 
-    static void verifyDirectoryContainsInstallation(Path path) throws ArgumentParsingException {
-        File dotGalleonDir = path.resolve(InstallationMetadata.GALLEON_INSTALLATION_DIR).toFile();
-        File channelsFile = path.resolve(ProsperoMetadataUtils.METADATA_DIR)
-                .resolve(ProsperoMetadataUtils.INSTALLER_CHANNELS_FILE_NAME).toFile();
-        if (!dotGalleonDir.isDirectory() || !channelsFile.isFile()) {
-            if (currentDir().equals(path)) {
-                // if the path is the current directory, user may have forgotten to specify the --dir option
-                //   -> suggest to use --dir
-                throw CliMessages.MESSAGES.invalidInstallationDirMaybeUseDirOption(path);
-            } else {
-                throw CliMessages.MESSAGES.invalidInstallationDir(path);
+    static Path determineInstallationDirectory(Optional<Path> directoryOption, Path currentDir) throws ArgumentParsingException {
```

Comment:
I think this can be private, testing can happen through the `determineInstallationDirectory` method or am I missing something?

Reply:
*This* is the `determineInstallationDirectory()` method. It now has two variants, one that only takes only `Optional<Path> directoryOption` parameter, and the second that takes `Optional<Path> directoryOption, Path currentDir` parameters.

The second parameter in the latter method is to make testing easier because otherwise it's not very well possible to set working directory inside a java process, that's why it's given as a method parameter.

So we need this latter method to be package private to make it visible for unit test.

Reply:
Good point I was thinking about it backwards